### PR TITLE
 Completed adjustments as needed for contact page.

### PR DESCRIPTION
**Pull request by ocampd1**

### CODE REVIEW

File: bp-web-client/angular-app/src/app/app.module.ts [hunk 1]
```diff
@@ -27,6 +27,8 @@ import {ServicesComponent} from './components/app-services/app-services.componen
 import {PhoneNumberDirective} from './directives/phone-number.directive';
 import {AppRoutingModule} from './app-routing.module';
 import {AppRoutingService} from './shared/services/app-routing.service';
+import { CommonModule } from '@angular/common';
+import {ContactButtonService} from './services/contact-button.service';
```
Comment (pouncilt) [inline bp-web-client/angular-app/src/app/app.module.ts:31]: Spacing here.

**Review by pouncilt — CHANGES_REQUESTED** [review]

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply package-lock.json:1]: Was not actually included in any of the files. Decided to delete it instead

File: package.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply package.json:1]: Was not actually included in any of the files. Decided to delete it instead

File: bp-web-client/angular-app/src/app/components/app-home/app-home.component.ts [hunk 4]
```diff
@@ -1,18 +1,19 @@
 import {AfterViewInit, Component, ElementRef, OnDestroy, OnInit, ViewChild} from '@angular/core';
 import * as $ from 'jquery';
-import {fromEvent, Observable, Subscription} from "rxjs";
+import {fromEvent, Observable, Subscription} from 'rxjs';
 import {ScrollService} from '../../services/scroll.service';
-
+import {environment} from '../../../environments/environment';
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/components/app-home/app-home.component.ts:5]: Added spacing.

File: bp-web-client/angular-app/src/app/components/app-work/app-work.component.ts [hunk 5]
```diff
@@ -1,12 +1,13 @@
 import { Component, OnInit } from '@angular/core';
+import {environment} from '../../../environments/environment';
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/components/app-work/app-work.component.ts:2]: Added spacing.

File: bp-web-client/angular-app/src/app/services/contact-button.service.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class ContactButtonService {
+  notOnContact = true;
+  onContact = false;
+
+  constructor() { }
+
+public OnContactPage(notOnContact) {
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/services/contact-button.service.ts:12]: refactored method to isOnContactView(notOnContactView: boolean): boolean

File: bp-web-client/angular-app/src/app/components/app-contact/app-contact.component.ts [hunk 7]
```diff
@@ -27,7 +28,8 @@ export class ContactComponent implements OnInit {
   constructor(private quoteService: QuoteService,
               private spinner: NgxSpinnerService,
               private appAlertOverlayModalService: AppAlertOverlayModalService,
-              public scrollTo: ScrollService) {
+              public scrollTo: ScrollService,
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/components/app-contact/app-contact.component.ts:31]: renamed to scrollToService

File: bp-web-client/angular-app/src/app/components/app-contact/app-contact.component.ts [hunk 8]
```diff
@@ -27,7 +28,8 @@ export class ContactComponent implements OnInit {
...
+              public onContact: ContactButtonService) {
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/components/app-contact/app-contact.component.ts:32]: renamed to contactButtonService

File: bp-web-client/angular-app/src/app/shared/components/app-footer/app-footer.component.html [hunk 9]
```diff
@@ -13,24 +13,34 @@ <h3><a href="mailto:info@bytepushers.software">info@bytepushers.software</a></h3
         </ul>
     <ul>
       <li><h1>Our Recent Projects</h1></li>
-      <li><h3><a href="https://www.chuckspick3.com">www.chuckspick3.com</a></h3></li>
+      <li><h3><a href="{{chucksPick3Url}}">www.chuckspick3.com</a></h3></li>
     </ul>
 
       </div>
-      <div class="col-sm-4 footer-quote" style=" height: 120px;">
+      <div class="col-sm-4 footer-quote" style=" height: 120px;" *ngIf="this.isOnContact.notOnContact">
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/shared/components/app-footer/app-footer.component.html:20]: omitted this keyword entirely.

File: bp-web-client/angular-app/src/app/shared/components/app-footer/app-footer.component.html [hunk 10]
```diff
@@ -43,20 +53,31 @@ <h3 ><a href="mailto:info@bytepushers.software" class="mobile">info@bytepushers.
         </ul>
         <ul>
           <li><h1 class="mobile">Our Recent Projects</h1></li>
-          <li><h3><a href="https://www.chuckspick3.com" class="mobile">www.chuckspick3.com</a></h3></li>
+          <li><h3><a href="{{chucksPick3Url}}" class="mobile">www.chuckspick3.com</a></h3></li>
         </ul>
 
       </div>
-      <div class="col-sm-4 footer-quote-mobile" style=" height: 120px;">
+      <div class="col-sm-4 footer-quote-mobile" style=" height: 120px;" *ngIf="this.isOnContact.notOnContact">
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/shared/components/app-footer/app-footer.component.html:60]: omitted .this keyword entirely.

File: bp-web-client/angular-app/src/app/shared/components/app-footer/app-footer.component.ts [hunk 11]
```diff
@@ -1,31 +1,38 @@
...
 @Component({
   selector: 'app-footer',
   templateUrl: './app-footer.component.html',
   styleUrls: ['./app-footer.component.css']
 })
 export class AppFooterComponent implements OnInit {
-
-  constructor() { }
+  public chucksPick3Url = environment.CHUCKS_PICK_3_URL;
+  constructor(public isOnContact: ContactButtonService,
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/shared/components/app-footer/app-footer.component.ts:13]: Made variables private and applied getters and setters for component.

File: bp-web-client/angular-app/src/app/app.module.ts [hunk 12]
```diff
@@ -57,12 +59,15 @@ import {AppRoutingService} from './shared/services/app-routing.service';
...
-    NgxPageScrollModule
+    NgxPageScrollModule,
+    CommonModule
   ],
   providers: [
     RouterLinkActive,
     QuoteService,
     {provide: Window, useValue: window},
+    ContactButtonService,
+    {provide: Window, useValue: window},
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/app.module.ts:70]: removed line.

File: bp-web-client/angular-app/src/app/app.module.ts (hunk 1 above)
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/app.module.ts:31]: Added spacing

File: bp-web-client/angular-app/src/app/services/scroll.service.ts [hunk 13]
```diff
@@ -1,11 +1,12 @@
 import { Injectable } from '@angular/core';
+import {ContactButtonService} from './contact-button.service';
 
 @Injectable({
   providedIn: 'root'
 })
 export class ScrollService {
 
-  constructor() { }
+  constructor(public onContact: ContactButtonService) { }
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/services/scroll.service.ts:9]: removed from constructor, not used at all in service.

File: bp-web-client/angular-app/src/app/services/scroll.service.ts [hunk 14]
```diff
@@ -1,11 +1,12 @@
 import { Injectable } from '@angular/core';
+import {ContactButtonService} from './contact-button.service';
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/services/scroll.service.ts:2]: Removed from service, not used at all.

File: bp-web-client/angular-app/src/app/shared/components/app-footer/app-footer.component.ts [hunk 15]
```diff
@@ -1,31 +1,38 @@
...
+              public scrollTo: ScrollService) { }
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/shared/components/app-footer/app-footer.component.ts]: Renamed to ScrollToService.

File: bp-web-client/angular-app/src/app/shared/components/app-footer/app-footer.component.html [hunk 16]
```diff
@@ -43,20 +53,31 @@ <h3 ><a href="mailto:info@bytepushers.software" class="mobile">info@bytepushers.
...
         <ul>
           <li><h1 class="mobile">Let's Work Together!</h1></li>
           <li style="margin-top: 5%">
             <a class="lets-talk-button" [routerLink]="'/contact'" (click)="backToTop()"  style="width: 100%">Request A Quote</a>
           </li>
         </ul>
       </div>
+      <div class="col-sm-4 footer-quote-mobile" style=" height: 120px;" *ngIf="this.isOnContact.onContact">
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply bp-web-client/angular-app/src/app/shared/components/app-footer/app-footer.component.html]: done

**Review by pouncilt — APPROVED** [review]

**Pull request merged**